server/src/server.js

```diff
@@ -694,11 +724,15 @@ app.post("/api/set-title/:id/:domain", csrfProtection, function(req, res) {
     simpleResponse(res, "Not logged in", 401);
     return;
   }
-  Shot.get(req.backend, shotId, req.deviceId).then((shot) => {
+  Shot.get(req.backend, shotId).then((shot) => {
     if (!shot) {
       simpleResponse(res, "No such shot", 404);
       return;
     }
+    if (req.accountId && shot.accountId != req.accountId) {
```

Comment:
It looks like this replaces the deviceId permission check with an accountId permission check – but as a result drops the deviceId check so that a guest user can't edit their title.